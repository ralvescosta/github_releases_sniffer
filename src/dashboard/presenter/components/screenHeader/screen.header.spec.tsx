import React from 'react';
import {render} from '@testing-library/react-native';

import {Header} from './index';

function makeSut(account: any = {}, amountOfAlerts: number = 1, amountOfRepos: number = 1, isNetworkAvailable: boolean = true) {
  return render(
    <Header account={account} amountOfAlerts={amountOfAlerts} amountOfRepos={amountOfRepos} isNetworkAvailable={isNetworkAvailable} />,
  );
}

describe('Screen Header Components', () => {
  it('Should render correctly', () => {
    const sut = makeSut();

    expect(sut.toJSON()).toMatchSnapshot();
  });
});
