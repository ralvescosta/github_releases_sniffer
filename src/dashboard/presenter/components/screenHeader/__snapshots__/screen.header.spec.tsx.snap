// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Header Components Should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#24292e",
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 22.5,
      "paddingVertical": 11.5,
      "width": "100%",
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 15,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#e1e4e8",
            "marginLeft": 26.5,
          }
        }
      >
        Configured Repos: 
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 15,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#e1e4e8",
            "marginLeft": 26.5,
          }
        }
      >
        Alerts last Week: 
        1
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 26.5,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 26.5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#e1e4e8",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
